server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java

```diff
@@ -299,4 +299,28 @@ void givenTwoTaskManagerAndATaskRanPerTaskManagerListingThemOnEachShouldShowBoth
                 });
         }
     }
+
+    @Test
+    void givenATaskManagerItShouldNotKeepExecutingATaskIfTheWorkqueueIsClosed(CountDownLatch countDownLatch) throws Exception {
+        try (EventSourcingTaskManager taskManager1 = taskManager()) {
+            TaskId firstTask = taskManager1.submit(new MemoryReferenceTask(() -> {
+                countDownLatch.await();
+                return Task.Result.COMPLETED;
+            }));
+            awaitUntilTaskHasStatus(firstTask, TaskManager.Status.IN_PROGRESS, taskManager1);
+
+            TaskId taskToExecuteAfterFirstNodeIsDown = taskManager1.submit(new CompletedTask());
+            //wait some time to let the faulty behavior happens. ie : a task will be `enqueued` in the worker
+            // if the workqueue use a parallel operator like `flatMap` on the flux.
+            Thread.sleep(200);
```

Comment:
Not really because, in the happy path when the bug is solved. there is no need for the sleep and no way to check for this.
The sleep is only to make sure that if we use a bad operator, the workqueue has the time to send another task which will be "enqueued` (waiting for the semaphore release) in the worker.

Reply:
I think that the test in itself is not testing what we want. We want to make sure the single active consumer will never dequeue more than one task from rabbit.
AFAIU, when we use flatmap, il has the effect to dequeue several tasks and put them in a Flux managed queue in memory.
If my description is correct, we can put 2 tasks in the queue, consume the first, stop the worker and ensure that the second task is still in the queue.
WDYT?

Reply:
I showed that the semaphore in the worker was just a workaround this issue.